### PR TITLE
Add suggestions/fixes to skeleton

### PR DESCRIPTION
**Pull request by Lobarr**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [] My pull request targets the `master` branch of Chapter.

Closes #XXXXX

Description of Changes
- added speccy to dev deps 
- added / updated scripts
- fixed husky deprecation warnings  
- setup jest and added scripts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,15 +7,16 @@
     "doc": "docs"
   },
   "scripts": {
-    "speccy": "speccy serve api/swagger.json",
-    "prod:server": "NODE_ENV=production ts-node --files --transpile-only ./server/index.ts",
     "dev:server": "ts-node --files  --transpile-only --project ./tsconfig.server.json ./server/index.ts",
-    "dev": "./node_modules/nodemon/bin/nodemon.js --watch 'server/**/*.ts' --exec 'npm run dev:server'",
-    "start": "npm run prod:server",
-    "nodemon": "./node_modules/nodemon/bin/nodemon.js",
+    "dev": "npx nodemon --watch 'server/**/*.ts' --exec 'npm run dev:server'",
```
Comment (timmyichen) [inline package.json:11]: any particular reason to npx nodemon instead of using the one installed as a dep?

Comment (Lobarr) [reply package.json:11]: In this case npx gets the nodemon binary from the node_module folder (installed deps) hence I find using npx cleaner than calling the binary manually.

Comment (timmyichen) [reply package.json:11]: Oh neat I didn't know it worked like that - thanks! Otherwise lgtm, will merge when at computer

Comment (timmyichen) [reply package.json:11]: Ah can you resolve the conflicts? Probably just new packages

File: package.json [hunk 2]
```diff
@@ -86,12 +90,6 @@
...
   "engines": {
     "node": "10.x",
-    "npm": ">= 6.0.0",
-    "speccy": "^0.11.0"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "lint-staged"
-    }
+    "npm": ">= 6.0.0"
```
Comment (timmyichen) [inline package.json]: Whoop this should be back in engines 🙃

File: package.json (hunk 1 above)
Comment (Lobarr) [reply package.json:11]: Resolved them

**Review by timmyichen — APPROVED** [review]
Thanks! :) gonna make some minor edits as a separate commit (separate command for building docker) but otherwise all is good!

**Pull request merged**